Contributors-Guide/GlusterFS-Release-process.md

```diff
@@ -1,28 +1,116 @@
+
+Minor releases will have guaranteed backwards compatibilty with earlier minor releases of the same branch.
+
+- LTM and STM state alternates with every release.
+- LTM and STM releases differ in the period of updates provided.
+
+### Long Term Maintenance (LTM) releases
+An LTM release reaches EOL when a LTM+2 release is done, 1 year later. LTM releases may get monthly minor releases for bug-fixes.
+
+### Short Term Maintenance (STM) releases
+STM releases are only supported until the next major release. STM releases may get monthly updates for bug fixes.
+
+The discussions around release schedules and lifecycles happened in the following mail threads on the GlusterFS maintainers mailing list.
+
+- https://www.gluster.org/pipermail/maintainers/2016-June/000892.html
+- https://www.gluster.org/pipermail/maintainers/2016-August/001174.html
+
+
+## GlusterFS major release window
+Each GlusterFS major release has a 3 month release window, in which changes get merged. The 3 month release window is split into two phases.
+
+1. A Open phase, where all changes get merged
+1. A Stability phase, where only changes that stabilize the release get merged.
+
+The first 2 months of a release window will be the Open phase, and the last month will be the stability phase.
+
+The release engineer (or team doing the release) is responsible for messaging.
+
+#### Open phase
+Any changes are accepted during this phase. New features that are introduced in this phase, need to be capable of being selectively built. All changes in the master branch are automatically incuded in the next release.
+
+All changes will be accepted during the Open phase. The changes have a few requirements,
+
+- a change fixing a bug SHOULD have public test case
+- a change introducing a new feature MUST have a disable switch that can disable the feature during a build
+
+
+#### Stability phase
+This phase is used to stabilize any new features introduced in the open phase, or general bug fixes for already existing features. A new release-<version> branch is created, all changes need to be sent to the master branch before getting backported to the new release branch. [kshlm: I prefer that the release gets tagged and branched at the end of the stability phase. This reduces the need for backports.]
+
+No new features will be merged in this phase.
 
-Notify packagers
-----------------
+At the end of this phase, any new feature introduced during the Open phase, that hasn't been declared stable, will be disabled, if possible removed to prevent confusion and set clear expectations towards users and developers.
 
-Notify the packagers that we need packages created. Provide the link to the
-source tarball from the Jenkins release job to the [packagers
-mailinglist](mailto:packaging@gluster.org). A list of the people involved in
-the package maintenance for the different distributions is in the `MAINTAINERS`
-file in the sources.
+The branching of the release happens at the end of this phase.
```

Comment:
On Fri, Aug 05, 2016 at 02:25:02AM -0700, Aravinda VK wrote:
> >  
> > -Notify the packagers that we need packages created. Provide the link to the
> > -source tarball from the Jenkins release job to the [packagers
> > -mailinglist](mailto:packaging@gluster.org). A list of the people involved in
> > -the package maintenance for the different distributions is in the `MAINTAINERS`
> > -file in the sources.
> > +The branching of the release happens at the end of this phase.
> 
> Branching should happen before this phase, at the end of Open phase. We can't block the patches in mainline because of any release.

Indeed, we should not.